### ISSUE TITLE
Always convert to ZonedDateTime before formatting, or provide methods for conversion

### ISSUE THREAD

**Issue by pcroc** [thymeleaf/thymeleaf-extras-java8time#10]
It's a bit odd that calling format() without a format string causes the temporal to be converted to ZonedDateTime, but if a format string is passed then no conversion occurs.

I'm trying format an Instant into a date string, but Instant only supports seconds and smaller fields. Therefore I have to do something horrible like this:

```
${#temporals.format(#temporals.createDate(report.getPublishedOn().toString().substring(0, 10)), 'dd MMM yyyy')}
```

It would be nice to have conversion methods exposed by the utility, or even better have it always convert to ZonedDateTime - particularly for Instant which is otherwise unusable.

**Comment by jmiguelsamper** [comment]
Hello Paul,

as far as I see, no conversion is done in any of the cases.

What is different is that format() uses a DateTimeFormatter with value

``` java
   new DateTimeFormatterBuilder().appendInstant().toFormatter()
```

whereas format(pattern) uses a DateTimeFormatter with value

``` java
   DateTimeFormatter.ofPattern(pattern, locale)
```

Please take a look at https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#ISO_INSTANT "an Instant cannot be formatted as a date or time without providing some form of time-zone"

I think you can format your Instant object as long as you provide a timezome to it, for example by appending

``` java
    instant.atZone(ZoneOffset.UTC);
```

This cannot be done automatically by a formatter build from a patter, but it can be done by the formater formatter (from DateTimeFormatterBuilder()) as it defaults to ZoneOffset.UTC.

**Comment by pcroc** [comment]
Ok it looks like this may have been changed following 2.1.0.RELEASE (commit b976509d39462398225bf85a6e2044e96e75d6b5).

Using `instant.atZone` is a good solution; it would helpful for the temporals dialect to provide easy access to the default zone, so I don't need to pass it in as a context variable.

Thanks.

**Comment by jmiguelsamper** [comment]
Oh, indeed I only looked at 3.0-dev code. It should be differences on that between latest 2.1-master and 3.0-dev branches.

**Comment by jmiguelsamper** [comment]
As the issue does not happen on the 3.0 branch, we won't fix it but feel free to send a PR for the 2.1 branch if you want.
